systems/controllers/control_utils.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <memory>
+#include <Eigen/Dense>
+
+#include "drake/common/trajectories/piecewise_polynomial.h"
+#include "drake/common/trajectories/exponential_plus_piecewise_polynomial.h"
+#include "drake/common/copyable_unique_ptr.h"
+
+namespace dairlib {
+namespace systems {
+
+/// ImposeHalfplaneGaurd() updates the foot placement position by restricting
+/// it in a halfplane.
+/// The halfplane is defined by two things:
+///  - A point on the line (the edge of the halfplace).
+///    The point is either shifted CoM or stance foot depending on the motion of
+///    the robot. This CoM is shifted by a distance `center_line_offset` in
+///    order to avoid foot collision.
+///  - The slope/direction of the line.
+///    The direction of the line is the robot's yaw angle.
+///
+/// Inputs:
+///  - `foot_placement_pos` planned foot placement position
+///  - `CoM` center of mass position
+///  - `left_stance_state` left stance state (of finite state machine)
+///  - `right_stance_state` right stance state (of finite state machine)
+///  - `state` current state of finite state machine
+///  - `yaw` current yaw angle
+///  - `CoM` current center of mass position
+///  - `stance_foot_pos` current stance foot position
+///  - `center_line_offset` offset of the center line
+///
+/// Requirement:
+///  - The controller is used with two-state finite state machine
+Eigen::Vector2d ImposeHalfplaneGaurd(Eigen::Vector2d foot_placement_pos,
```

Comment:
Actually this can be used more broadly. The only requirement for using this function is that the robot has to be bipedal. 
I'll replace `right/left stance state` and `state` with one boolean `left_right_stance_flag`.

Reply:
It's OK, but feels pretty specific. Fine to leave for now.